### PR TITLE
Adding append_polydata method

### PR DESCRIPTION
**Pull request by beroda**
### Overview

This PR originates from comments https://github.com/pyvista/pyvista/pull/4227#issuecomment-1573418296 and https://github.com/pyvista/pyvista/pull/4227#issuecomment-1573639737.

Since https://github.com/pyvista/pyvista/pull/4227 replaced  `vtkAppendPolyDataFilter` by `DataSetFilters.merge` in `PolyDataFilters.merge` to greatly improve performance and coherence with other merging operations, points ordering is altered and can cause some users to face issues.

This PR is **not** a roll-back of https://github.com/pyvista/pyvista/pull/4227 since it simply adds a `PolyDataFilter.append_polydata` method re-implementing `vtkAppendPolyDataFilter`.

A note regarding this change of behavior for `0.39.0` is also added in the documentation of `PolyDataFilters.merge` for users experiencing the same issues.

### Details
```python
import pyvista as pv
sp0 = pv.Sphere()
sp1 = sp0.translate((1, 0, 0))
appended = sp0.append_polydata(sp1)
appended.plot()

# Support for *args
sp2 = sp0.translate((-1, 0, 0))
appended = sp0.append_polydata(sp1, sp2)
appended.plot()
```

### CODE REVIEW

**Comment by tkoyama010** [comment]
Thank you for adding this. Please hold off merging until we merge https://github.com/pyvista/pyvista/pull/4486 and resolve the conflict with it.

File: pyvista/core/filters/poly_data.py
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:396]:
```suggestion
        pyvista.PolyData
            Appended PolyData(s).
```

File: pyvista/core/filters/poly_data.py
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:457]:
```suggestion
            To enhance performance and coherence with merging operations available for other datasets in pyvista,
```

File: pyvista/core/filters/poly_data.py
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:459]:
```suggestion
            irrespectively of input datasets types.
```

**Review by banesullivan — APPROVED** [review]
Well done! Thank you!

**Pull request merged**
